rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/values.kt

```diff
@@ -209,6 +210,7 @@ data class StringValue(var value: String, var varying: Boolean = false) : Abstra
         when (other) {
             is StringValue -> compare(other, DEFAULT_CHARSET)
             is BlanksValue -> if (this.isBlank()) EQUAL else SMALLER
+            is BooleanValue -> if (this.value.isInt() && this.value.toInt() == 1) EQUAL else GREATER
```

Comment:
I'm not really sure about the correctness of your implementation, it seems you are taking in account only the case where BooleanValue is *ON.

Reply:
My bad, and it's so big, I forgot to replace `1` with the value of `other`